src/sat.cc

```diff
@@ -1810,6 +1886,10 @@ void Sat::DeleteThreads() {
   logprintf(12, "Log: Destroying WorkerStatus objects\n");
   power_spike_status_.Destroy();
   continuous_status_.Destroy();
+
+  // Delete thread test steps
+  for (std::unique_ptr<TestStep> &thread_step : thread_test_steps_)
```

Comment:
Will all the steps show up as completing at the same time? Does it make sense to pass the test steps to the threads themselves where they can complete and free up? Will have to create a step per thread like 'cpu_stress_thread_1_of_6'.

Reply:
The test steps are being passed to the threads themselves - the threads just don't own the steps (the unique ptr) because that pattern only works for some of the threads based on how their setup (for instance some threads spawn multiple child threads and can't pass ownership to all of them).

Right now each thread of the same type is getting a pointer to the same test step so they will add their artifacts to that same test step.

The steps will all end at the same time because they all do end at the same time.

Here's the flow of the SAT test:
1. Setup
2. Spawn all the threads (and their associated test steps)
3. wait 20 seconds for the threads to run.
4. Join the threads
5. Run analysis, which we turn into measurements that we add to the test steps
6. Delete the threads and end the test steps.

So the join step causes all the threads to stop running, ending them at the same time. We then to all the post analysis, and then delete all the threads and close the test steps.

Reply:
The multi-threading case needs more thought, because if there are multiple threads concurrently as part of the same step, the artifacts from the different threads will likely not have any disambiguating field and be mixed up. Some guidance in the standard is likely worthwhile.

Also, the join doesn't cause the threads to stop running AFAIK, it just waits for the threads to complete.

Reply:
Discussed offline, this is good for now while we figure out how to handle the multi threading portion.